src/Parser/Core/Modules/Haste.js

```diff
@@ -179,29 +185,29 @@ class Haste extends Analyzer {
   }
 
   _applyHasteGain(event, haste) {
-    const oldHaste = this.current;
-    this.current = this.constructor.addHaste(this.current, haste);
-
-    this._triggerChangeHaste(event, oldHaste, this.current, haste, null);
+    this._setHaste(event, this.constructor.addHaste(this.current, haste));
   }
   _applyHasteLoss(event, haste) {
+    this._setHaste(event, this.constructor.removeHaste(this.current, haste));
+  }
+  _setHaste(event, haste) {
     const oldHaste = this.current;
-    this.current = this.constructor.removeHaste(this.current, haste);
+    this.current = haste;
 
-    this._triggerChangeHaste(event, oldHaste, this.current, null, haste);
+    this._triggerChangeHaste(event, oldHaste, this.current);
   }
-  _triggerChangeHaste(event, oldHaste, newHaste, hasteGain, hasteLoss) {
-    this.owner.triggerEvent('changehaste', {
+  _triggerChangeHaste(event, oldHaste, newHaste) {
+    const fabricatedEvent = {
       timestamp: event ? event.timestamp : this.owner.currentTimestamp,
       type: 'changehaste',
       sourceID: event ? event.sourceID : this.owner.playerId,
       targetID: this.owner.playerId,
       reason: event,
       oldHaste,
       newHaste,
-      hasteGain,
-      hasteLoss,
```

Comment:
Just to be completely sure, did you do a search to check that these fields aren't used anywhere?

Reply:
Yes